File: 백준/Silver/1620. 나는야 포켓몬 마스터 이다솜/나는야 포켓몬 마스터 이다솜.swift
let nm = readLine()!.split(separator: " ").map { Int($0)! }
let n = nm[0], m = nm[1]
var idxDict = [Int: String]()
var nameDict = [String: Int]()
var output = ""

for i in 1...n {
    let name = readLine()!
    idxDict[i] = name
    nameDict[name] = i
}
for _ in 0..<m {
    let input = readLine()!
    if let num = Int(input) {
        output += "\(idxDict[num]!)\n"
    } else {
        output += "\(nameDict[input]!)\n"
    }
}
print(output, terminator: "")

File: 백준/Silver/1620. 나는야 포켓몬 마스터 이다솜/나는야 포켓몬 마스터 이다솜.py
toString = {}
toInt = {}

(n, m) = map(int, input().split())

for i in range(1, n + 1):
    name = input()
    toString[i] = name
    toInt[name] = i

for _ in range(0, m):
    line = input()
    if line[0] >= 'A' and line[0] <= 'Z':
        print(toInt[line])
    else:
        print(toString[int(line)])